### PR TITLE
Transition away blank Ouput:Table:SummaryReport report names in v8.8 to v8.9 transition

### LINKED ISSUE #6919: Blank report name in Output:Table:SummaryReports results in fatal error with no error message identifying the problem
Issue overview
--------------
This object
```
Output:Table:SummaryReports,
    AnnualBuildingUtilityPerformanceSummary,  !- Report 1 Name
    ZoneHeatingSummaryMonthly,  !- Report 2 Name
    ZoneElectricSummaryMonthly,  !- Report 3 Name
    ZoneCoolingSummaryMonthly,  !- Report 4 Name
    SensibleHeatGainSummary, !- Report 5 Name
    ,!- Report 6 Name
    SpaceGainsMonthly,       !- Report 7 Name
    EnvelopeSummary,         !- Report 8 Name
    ObjectCountSummary,      !- Report 9 Name
    ComponentCostEconomicsSummary,  !- Report 10 Name
    AllSummary;              !- Report 11 Name
```
results in this error file
```
Program Version,EnergyPlus, Version 8.9.0-40101eaafd, YMD=2018.08.21 10:40,
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  0.14sec
```

Help desk ticket 13045 has blank meter names in Output:Table:Monthly and Output:Table:Annual

Output:Table:Monthly,
  Overall HVAC System Energy,  !- Name
  3,                       !- Digits After Decimal
  Cooling:Electricity,     !- Variable or Meter 1 Name
  SumOrAverage,            !- Aggregation Type for Variable or Meter 1
  ,                        !- Variable or Meter 2 Name
  SumOrAverage;            !- Aggregation Type for Variable or Meter 2

Output:Table:Annual,
  OUTPUT TABLE 1,          !- Name
  ,                        !- Filter
  ,                        !- Schedule Name
  ,                        !- Variable or Meter or EMS Variable or Field 1 Name
  ,                        !- Aggregation Type for Variable or Meter 1
  2,                       !- Digits After Decimal 1
  ,                        !- Variable or Meter or EMS Variable or Field 2 Name
  ,                        !- Aggregation Type for Variable or Meter 2
  2,                       !- Digits After Decimal 2
  ,                        !- Variable or Meter or EMS Variable or Field 3 Name
  ,                        !- Aggregation Type for Variable or Meter 3
  2,                       !- Digits After Decimal 3
  ,                        !- Variable or Meter or EMS Variable or Field 4 Name
  ,                        !- Aggregation Type for Variable or Meter 4
  2,                       !- Digits After Decimal 4
  ,                        !- Variable or Meter or EMS Variable or Field 5 Name
  ,                        !- Aggregation Type for Variable or Meter 5
  2,                       !- Digits After Decimal 5
  ,                        !- Variable or Meter or EMS Variable or Field 6 Name
  ,                        !- Aggregation Type for Variable or Meter 6
  2,                       !- Digits After Decimal 6
  ,                        !- Variable or Meter or EMS Variable or Field 7 Name
  ,                        !- Aggregation Type for Variable or Meter 7
  2,                       !- Digits After Decimal 7
  ,                        !- Variable or Meter or EMS Variable or Field 8 Name
  ,                        !- Aggregation Type for Variable or Meter 8
  2,                       !- Digits After Decimal 8
  ,                        !- Variable or Meter or EMS Variable or Field 9 Name
  ,                        !- Aggregation Type for Variable or Meter 9
  2,                       !- Digits After Decimal 9
  ,                        !- Variable or Meter or EMS Variable or Field 10 Name
  ,                        !- Aggregation Type for Variable or Meter 10
  2,                       !- Digits After Decimal 10
  ,                        !- Variable or Meter or EMS Variable or Field 11 Name
  ,                        !- Aggregation Type for Variable or Meter 11
  2,                       !- Digits After Decimal 11
  ,                        !- Variable or Meter or EMS Variable or Field 12 Name
  ,                        !- Aggregation Type for Variable or Meter 12
  2,                       !- Digits After Decimal 12
  ,                        !- Variable or Meter or EMS Variable or Field 13 Name
  ,                        !- Aggregation Type for Variable or Meter 13
  2,                       !- Digits After Decimal 13
  ,                        !- Variable or Meter or EMS Variable or Field 14 Name
  ,                        !- Aggregation Type for Variable or Meter 14
  2,                       !- Digits After Decimal 14
  ,                        !- Variable or Meter or EMS Variable or Field 15 Name
  ,                        !- Aggregation Type for Variable or Meter 15
  2;                       !- Digits After Decimal 15
Error file results:

Program Version,EnergyPlus, Version 9.0.0-63db0a5a1f, YMD=2018.08.24 09:46,
 ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
 ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
 ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
 ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
 ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min 11.61sec

### Details
Some additional details for this issue (if relevant):
 - Win64
 - Version of EnergyPlus v8.9.0
 - Unmethours [33505](https://unmethours.com/question/33505/energyplus-89-crashes-after-using-updated-88-idf-file/?comment=33608#post-id-33608)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file attached here 
[6919-1ZoneUncontrolled-v89.idf.txt](https://github.com/NREL/EnergyPlus/files/2307019/6919-1ZoneUncontrolled-v89.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #6919
Starting with v8.9, blank fields in Output:Table:SummaryReports result in a fatal error. Transition-V8-8-0-to-V8-9-0 will not remove any blank fields in this object.

Defect files are attached here:
[6919-5ZoneDDCycOnAny-V880.idf.txt](https://github.com/NREL/EnergyPlus/files/4921087/6919-5ZoneDDCycOnAny-V880.idf.txt)
[6919-5ZoneDDCycOnAny-NoBlanks_V880.idf.txt](https://github.com/NREL/EnergyPlus/files/4921083/6919-5ZoneDDCycOnAny-NoBlanks_V880.idf.txt )- to confirm this change doesn't break an object without blanks.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] ~Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair~
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] ~If any diffs are expected, author must demonstrate they are justified using plots and descriptions~
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions - file with new transition actually runs in v8.9 and beyond
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] ~If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label~
 - [ ] ~If structural output changes, add to output rules file and add OutputChange label~
 - [ ] ~If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies~

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [ ] CI status: all green or justified
 - [ ] ~Check that performance is not impacted (CI Linux results include performance check)~
 - [ ] ~Run Unit Test(s) locally~
 - [ ] ~Check any new function arguments for performance impacts~
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

### CODE REVIEW

**Comment by mjwitte** [comment]
@Myoldmopar This is ready for review. SolarShadingTestGPU diffs are unrelated.

File: src/Transition/CreateNewIDFUsingRulesV8_9_0.f90 [hunk 1]
```diff
@@ -810,6 +810,21 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
...
+                   ! Skip blank fields
+                   IF (.NOT. InArgs(TempArgsNum) == Blank) THEN
+                     CurField = CurField + 1
+                     OutArgs(CurField) = InArgs(TempArgsNum)
+                   END IF
+                 END DO
+                 IF (CurField < CurArgs) THEN
+                   Nodiff=.false.
+                   CurArgs = CurField
+                 END IF
```
Comment (Myoldmopar) [inline src/Transition/CreateNewIDFUsingRulesV8_9_0.f90:826]: OK, so just loop over the entire set of old fields, and if any blanks are there, skip them and mark the file as dirty.  No problem with this.

**Review by Myoldmopar — APPROVED** [review]
Will build and test transitioning files with and without blanks locally.

**Comment by Myoldmopar** [comment]
Ran both test files.  In the no-blanks case, the only diffs between 8.8 and 8.9 are the version number and some fields in the hvac equipment lists.  In the with-blanks case, those diffs are accompanied by blanks in the 8.8 file being cleaned out.  This is good to go.  Thanks @mjwitte!

**Pull request merged**